vcd/resource_vcd_vapp_vm.go

```diff
@@ -241,6 +242,104 @@ func resourceVcdVAppVm() *schema.Resource {
 				Optional: true,
 				Set:      resourceVcdVmIndependentDiskHash,
 			},
+			"override_template_disk": {
+				Type:        schema.TypeSet,
+				Optional:    true,
+				ForceNew:    true,
+				Description: " A block to match internal_disk interface in template. Multiple can be used. Disk will be matched by bus_type, bus_number and unit_number.",
+				Elem: &schema.Resource{Schema: map[string]*schema.Schema{
+					"bus_type": {
+						Type:         schema.TypeString,
+						Required:     true,
+						ForceNew:     true,
+						ValidateFunc: validation.StringInSlice([]string{"ide", "parallel", "sas", "paravirtual", "sata"}, true),
+						Description:  "The type of disk controller. Possible values: ide, parallel( LSI Logic Parallel SCSI), sas(LSI Logic SAS (SCSI)), paravirtual(Paravirtual (SCSI)), sata",
+					},
+					"size_in_mb": {
```

Comment:
Why is this property required? Isn't the size defined in the internal disk resource?
What happens if the sizes don't match?

Reply:
override template disk allows to change exisiting template disk before first VM boot up. Size in mb allows you increase size of disk existing in template.